src/AspNetCoreModuleV2/AspNetCore/Inc/applicationinfo.h

```diff
@@ -2,9 +2,14 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 #pragma once
-#define API_BUFFER_TOO_SMALL 0x80008098
 
-extern BOOL     g_fRecycleProcessCalled;
+#include "precomp.hxx"
```

Comment:
why do we need to add those includes here?

Reply:
So we see what every file uses and can control it, also makes every header file self contaned.

>We used centralized precompiled header file to avoid adding multiple includes in each individual file and issue of orders

It cause us to have a massive amount of unneeded included headers. There are no ordering issues when every header includes only what it needs. There is a single case of circular dependency that is fixed by forwarding definition.

Reply:
Circular dependencies are the biggest reason for this change IMO. I've battled a few not recognizing why I caused a circular dependency. It also makes it very clear what classes depend on what.

Reply:
Another benefit is that we don't need to figure out how to order headers in precomp correctly so they compile.

Reply:
I believe the old IIS team use precompiled header file based on 1) less and clean code and 2) less compilation time. Circular dependency is a code design issue instead of compilation, which could happen even you put include into every header files. I personally still prefer the precompiled header as it compilation efficient and help to identify code issue. This is why VS c+ template use precompiled header by default.

Reply:
Precompiled headers are not enabled in our solution at all. We just use it as an include directive dump and gain no perf.

Additionally, precompiled headers are not supposed to be used for header files from current project at all because it causes precompiled header invalidation every time the code is changed ruining the entire purpose.
Precompiled headers are designed to hold large external includes that rarely change.

> help to identify code issue

Which code issues does it help to identify?